erpnext/accounts/utils.py

```diff
@@ -730,6 +730,22 @@ def update_reference_in_payment_entry(
 	}
 	update_advance_paid = []
 
+	# Update Reconciliation effect date in reference
+	if payment_entry.advance_reconciliation_takes_effect_on == "Advance Payment Date":
```

Comment:
Handle this only for "Advance Payments". Normal payments shouldn't have any `reconcile_effect_on`